x-pack/plugins/apm/common/agent_configuration/amount_and_unit.ts

```diff
@@ -4,15 +4,15 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
-interface AmountAndUnit {
-  amount: string;
+export interface AmountAndUnit {
+  amount: number;
   unit: string;
 }
 
 export function amountAndUnitToObject(value: string): AmountAndUnit {
   // matches any postive and negative number and its unit.
   const [, amount = '', unit = ''] = value.match(/(^-?\d+)?(\w+)?/) || [];
-  return { amount, unit };
+  return { amount: parseInt(amount, 10), unit };
 }
 
 export function amountAndUnitToString({ amount, unit }: AmountAndUnit) {
```

Comment:
nit: We could allow `string` here and then avoid `parseInt` in x-pack/plugins/apm/public/components/app/Settings/AgentConfigurations/AgentConfigurationCreateEdit/SettingsPage/SettingFormRow.tsx
```suggestion
export function amountAndUnitToString({ amount, unit }: {
  amount: number | string;
  unit: string;
}) {
```